enzy_htp/preparation/reactive_docking.py

```diff
@@ -0,0 +1,1285 @@
+"""Driver for the reactive docking functionality available in enzy_htp. The only function that that should be 
+called is dock_reactants(). All others are implementation functions that SHOULD NOT be used. 
+
+Author: Chris Jurich <chris.jurich@vanderbilt.edu>
+Date: 2023-07-28
+"""
+import os
+from pathlib import Path
+from collections import defaultdict
+from typing import List, Tuple, Set
+from copy import deepcopy
+
+import numpy as np
+import pandas as pd
+
+from enzy_htp import interface, config, _LOGGER, binding_energy
+from enzy_htp._interface.rosetta_interface import RosettaCst
+
+import enzy_htp.chemical as chem
+from enzy_htp import mutation as mm
+from enzy_htp.structure import PDBParser, Structure
+from enzy_htp.core import file_system as fs
+
+#TODO(CJ): I should go through and use existing values if cache mode is selected
+#TODO(CJ): need to be able to specify the size of the grid
+
+
+def dock_reactants(structure: Structure,
+                   reactants: List[str],
+                   constraints: List[RosettaCst] = None,
+                   mutations: List[List[mm.Mutation]] = None,
```

Comment:
I still dont complete get the idea of how have mutation here is different from doing mutation first and then run EnzyRCD again.

Reply:
Dont need to address it here. We can chat about it later.

Reply:
It should be the same if you move the mutation component here and put mutate_stru() in the main script (of apps enabled by this module) after calling dock_reactants().

Reply:
It's also tricky to determine whether we want to mutate first or dock first. I think mutate first is a better idea as it is how nature works and there are extreme cases that a mutation would completely disable the old binding site.

Reply:
So this is here in the event that a constraint and mutation overlap. If that is the case then the system is re-run separately with a set of modified inputs.

Reply:
Cool. In this case, we can just leave the mutant input only used for that purpose.